src/main/java/seedu/address/logic/commands/NotificationCommand.java

```diff
@@ -0,0 +1,129 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.exceptions.NoUserSelectedException;
+
+/**
+ * Toggles the notifications on and off.
+ */
+//@Snookerballs
+public class NotificationCommand extends Command {
+    public static final String COMMAND_WORD = "notification";
+    public static final String COMMAND_ALIAS = "n";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Toggles notification messages. "
+            + "Parameters: "
+            + PREFIX_NOTIFICATION_TYPE + "[NOTIFICATION TYPE] "
+            + PREFIX_TOGGLE + "on/off \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NOTIFICATION_TYPE + "warning "
+            + PREFIX_TOGGLE + "off";
+
+    public static final String MESSAGE_SUCCESS = "%1 type notifications toggled %1";
+
+    public static final String OPTION_ON = "on";
+    public static final String OPTION_OFF = "off";
+
+    public static final String OPTION_WARNING = "warning";
+    public static final String OPTION_TIP = "tip";
+
+    private NotificationCommandDescriptor descriptor;
+    /**
+     * Creates an AddCommand to add the specified {@code Expense}
+     */
+    public NotificationCommand(NotificationCommandDescriptor descriptor) {
+        this.descriptor = descriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) throws CommandException, NoUserSelectedException {
+        requireNonNull(model);
+        boolean toggleOption = false;
+        if (descriptor.getToggle().get().equals(OPTION_ON)) {
+            toggleOption = true;
+        } else {
+            toggleOption = false;
+        }
+
+        if (descriptor.getNotificationType().isPresent()) {
+            toggleSpecificNotification(model, descriptor.getNotificationType().get(), toggleOption);
+        } else {
+            model.toggleBothNotification(toggleOption);
+        }
+        model.commitExpenseTracker();
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    /**
+     * Toggles a specific type of notification
+     * @param model to modify
+     * @param type of notification to generate
+     * @param option to toggle to
+     * @throws NoUserSelectedException
+     */
+    public void toggleSpecificNotification(Model model, String type, boolean option) throws NoUserSelectedException {
+        if (type.equals(OPTION_WARNING)) {
+            model.toggleWarningNotification(option);
+        } else if (type.equals(OPTION_TIP)) {
+            model.toggleTipNotification(option);
+        }
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof NotificationCommand); // instanceof handles nulls
+    }
+
+    /**
+     * Stores the variables of the NotificationCommand as a class
+     */
+    public static class NotificationCommandDescriptor {
```

Comment:
A public static nested class is usually a bad idea. Does this really need to be public? By using a public nested class you are breaking abstraction very badly since you are exposing the inner workings of the outer class by allowing some other class to initialize a `NotificationCommandDescriptor`